adapters/MyTarget/MyTargetAdapter/GADMAdapterMyTargetNative.m

```diff
@@ -0,0 +1,264 @@
+		_nativeAd = nil;
+	}
+}
+
+- (BOOL)isBannerAnimationOK:(GADMBannerAnimationType)animType
+{
+	return YES;
+}
+
+- (void)presentInterstitialFromRootViewController:(UIViewController *)rootViewController
+{
+	id<GADMAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+	[strongConnector adapter:self didFailAd:[GADMAdapterMyTargetUtils errorWithDescription:kGADMAdapterMyTargetErrorInterstitialNotSupported]];
+}
+
+- (void)getNativeAdWithAdTypes:(NSArray *)adTypes options:(NSArray *)options
+{
+	id<GADMAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+
+	NSUInteger slotId = [GADMAdapterMyTargetUtils slotIdFromCredentials:strongConnector.credentials];
+	guard(slotId > 0) else
+	{
+		MTRGLogError(kGADMAdapterMyTargetErrorSlotId);
+		[strongConnector adapter:self didFailAd:[GADMAdapterMyTargetUtils errorWithDescription:kGADMAdapterMyTargetErrorSlotId]];
+		return;
+	}
+	
+	_isContentAdRequested = [adTypes containsObject:kGADAdLoaderAdTypeNativeContent];
+	_isAppInstallAdRequested = [adTypes containsObject:kGADAdLoaderAdTypeNativeAppInstall];
+	_adTypesRequested = [adTypes componentsJoinedByString:@", "];
+
+	guard(_isContentAdRequested || _isAppInstallAdRequested) else
+	{
+		NSString *description = [NSString stringWithFormat:kGADMAdapterMyTargetErrorInvalidNativeAdType, _adTypesRequested];
+		MTRGLogError(description);
+		[strongConnector adapter:self didFailAd:[GADMAdapterMyTargetUtils errorWithDescription:description]];
+		return;
+	}
+
+	BOOL shouldDownloadImages = YES;
+	for (GADNativeAdImageAdLoaderOptions *imageOptions in options)
+	{
+		if ([imageOptions isKindOfClass:[GADNativeAdImageAdLoaderOptions class]] && imageOptions.disableImageLoading)
+		{
+			shouldDownloadImages = NO;
+			break;
+		}
+	}
+
+	_nativeAd = [[MTRGNativeAd alloc] initWithSlotId:slotId];
+	_nativeAd.delegate = self;
+	_nativeAd.autoLoadImages = shouldDownloadImages;
+	[GADMAdapterMyTargetUtils fillCustomParams:_nativeAd.customParams withConnector:strongConnector];
+	[_nativeAd.customParams setCustomParam:kMTRGCustomParamsMediationAdmob forKey:kMTRGCustomParamsMediationKey];
+
+	if (_isContentAdRequested && !_isAppInstallAdRequested)
+	{
+		[_nativeAd.customParams setCustomParam:kGADMAdapterMyTargetNativeAdTypeContent forKey:kGADMAdapterMyTargetNativeAdTypeKey];
+	}
+	else if (_isAppInstallAdRequested && !_isContentAdRequested)
+	{
+		[_nativeAd.customParams setCustomParam:kGADMAdapterMyTargetNativeAdTypeInstall forKey:kGADMAdapterMyTargetNativeAdTypeKey];
+	}
+	[_nativeAd load];
+}
+
+- (BOOL)handlesUserClicks
+{
+	return YES;
+}
+
+- (BOOL)handlesUserImpressions
+{
+	return YES;
+}
+
+#pragma mark - MTRGNativeAdDelegate
+
+- (void)onLoadWithNativePromoBanner:(MTRGNativePromoBanner *)promoBanner nativeAd:(MTRGNativeAd *)nativeAd
+{
+	id<GADMAdNetworkConnector> strongConnector = _connector;
+	MTRGLogInfo();
+	guard(strongConnector) else return;
+
```

Comment:
Please add a check before creating a Mediated Native Ad to check if all the always included assets mentioned [here](https://support.google.com/admob/answer/6240809) are available. If any of the required assets are not available, the adapter should fail the request.

Reply:
Requested changes has been made.